de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/filesystem/IntelliJProjectImplV2.java

```diff
@@ -103,6 +93,41 @@ public IntelliJProjectImplV2(@NotNull final Module module) {
         }
     }
 
+    /**
+     * Returns the content root of the given module.
+     * <p>
+     * This method is used to enforce the current restriction that shared
+     * modules must contain exactly one content root.
+     * </p>
+     *
+     * @param module the module to get the content root for
+     * @return the content root of the given module
+     * @throws IllegalArgumentException if the given module does not have
+     *                                  exactly one content root
+     */
+    @NotNull
+    private static VirtualFile getModuleContentRoot(
+        @NotNull
+            Module module) {
+
+        ModuleRootManager moduleRootManager = ModuleRootManager
+            .getInstance(module);
+
+        VirtualFile[] contentRoots = moduleRootManager.getContentRoots();
+
+        int numberOfContentRoots = contentRoots.length;
+
+        if (numberOfContentRoots != 1) {
+            throw new IllegalArgumentException(
+                "Modules shared with Saros currently must contain exactly one "
```

Comment:
If you try to build a user friendly error message you should not rely on the toString() method of the objects you are using to build the string. In the worst case you get just pointer references which are not helpful for a user.

Reply:
I generally try to test whether an object has a useful `toString()` implementation before including it in a log message or exception. (`Module.toString()` just return the module name, `VirtualFile.toString()` returns the paths.)

But you are right. At some point I will forget to check if there is a `toString()` implementation (or worse: an implementation gets removed afterwards). I will try to avoid it in the future.